commands/completion_test.go

```diff
@@ -0,0 +1,73 @@
+package commands_test
+
+import (
+	"bytes"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/sclevine/spec"
+	"github.com/sclevine/spec/report"
+	"github.com/spf13/cobra"
+
+	"github.com/buildpack/pack/commands"
+	"github.com/buildpack/pack/internal/mocks"
+	"github.com/buildpack/pack/logging"
+	h "github.com/buildpack/pack/testhelpers"
+)
+
+func TestCompletionCommand(t *testing.T) {
+	spec.Run(t, "Commands", testCompletionCommand, spec.Random(), spec.Report(report.Terminal{}))
+}
+
+func testCompletionCommand(t *testing.T, when spec.G, it spec.S) {
+	var (
+		command    *cobra.Command
+		logger     logging.Logger
+		outBuf     bytes.Buffer
+		packHome   string
+		missingDir string
+	)
+
+	it.Before(func() {
+		logger = mocks.NewMockLogger(&outBuf)
+		// the CompletionCommand calls a method on its Parent(), so it needs to have
+		// one.
+		command = &cobra.Command{}
+		command.AddCommand(commands.CompletionCommand(logger))
+		command.SetArgs([]string{"completion"})
+		var err error
+		packHome, err = ioutil.TempDir("", "")
+		h.AssertNil(t, err)
+		missingDir = filepath.Join(packHome, "not-found")
+	})
+
+	it.After(func() {
+		os.RemoveAll(packHome)
+	})
+
+	when("#CompletionCommand", func() {
+		when("PackHome directory does exist", func() {
+			it.Before(func() {
+				os.Setenv("PACK_HOME", packHome)
```

Comment:
Although the risk is minimal it does leak the environment variable to further tests. We should really be unsetting the env var in an `it.After`.

Reply:
Good catch, thanks!